app/components/frame-viewer.jsx

```diff
@@ -44,63 +44,78 @@ export default class FrameViewer extends React.Component {
         case 'image':
-          return (
-            <div className="subject-image-frame" >
-              <img
-                role="presentation"
-                ref="subjectImage"
-                className="subject pan-active"
-                src={src}
-                onLoad={this.handleLoad}
-                tabIndex={0}
-                onFocus={this.panZoom ? this.panZoom.togglePanOn : () => {}}
-                onBlur={this.panZoom ? this.panZoom.togglePanOff : () => {}}
-              />
-
-              {this.state.loading && (
-                <div className="loading-cover" style={this.constructor.overlayStyle} >
-                  <LoadingIndicator />
-                </div>
-              )}
-            </div>);
+          return ImageViewer;
         case 'video':
-          return (
-            <VideoPlayer src={src} type={type} format={format} frame={this.props.frame} onLoad={this.handleLoad} />
-          );
+          return VideoPlayer;
         case 'text':
-          return (
-            <TextViewer src={src} type={type} format={format} frame={this.props.frame} onLoad={this.handleLoad} />
-          );
+          return TextViewer;
         default:
           return null;
       }
     })(type);
 
     if (FrameWrapper) {
+      if (type === 'image') {
+        return (
+          <PanZoom ref={(c) => { this.panZoom = c; }} enabled={zoomEnabled} frameDimensions={this.state.frameDimensions}>
+            <FrameWrapper
+              frame={this.props.frame}
+              naturalWidth={this.state.frameDimensions.width || 0}
+              naturalHeight={this.state.frameDimensions.height || 0}
+              workflow={this.props.workflow}
+              subject={this.props.subject}
+              classification={this.props.classification}
+              annotation={this.props.annotation}
+              loading={this.state.loading}
+              preferences={this.props.preferences}
+              modification={this.props.modification || {}}
+              onChange={this.props.onChange}
+            >
+              <FileViewer
+                ref="subjectImage"
+                src={src}
+                type={type}
+                format={format}
+                frame={this.props.frame}
+                onLoad={this.handleLoad}
+                onFocus={this.panZoom ? this.panZoom.togglePanOn : () => {}}
+                onBlur={this.panZoom ? this.panZoom.togglePanOff : () => {}}
+              />
+            </FrameWrapper>
+          </PanZoom>
+        );
+      } else {
+        return (
+          <div className="frame-annotator">
+            <FileViewer
+              ref="subjectImage"
+              src={src}
+              type={type}
+              format={format}
+              frame={this.props.frame}
+              onLoad={this.handleLoad}
+              onFocus={this.panZoom ? this.panZoom.togglePanOn : () => {}}
+              onBlur={this.panZoom ? this.panZoom.togglePanOff : () => {}}
```

Comment:
Good point. It would be even better if we could figure out a way for `PanZoom` to add these props dynamically when it wraps the viewer component. I guess it could pass them down through the SVG component?